### PR TITLE
Rewards improvements.

### PR DESCRIPTION
**Pull request by konrad04-1989**

### CODE REVIEW

File: app/models/employee.rb [hunk 1]
```diff
@@ -12,6 +12,6 @@ class Employee < ApplicationRecord
   has_many :rewards, through: :orders
 
   def number_of_points
-    received_kudos.count - rewards.sum(:price).to_i
+    received_kudos.count - orders.sum(&:order_snapshot_price)
```
Comment (ciejj) [inline app/models/employee.rb]: why not? then you can get rid of the `delegate` in order model :)
```suggestion
    received_kudos.count - orders.sum(&:order_snapshot.price)
```

File: app/models/order.rb [hunk 2]
```diff
@@ -1,6 +1,10 @@
 # frozen_string_literal: true
 
 class Order < ApplicationRecord
+  serialize :order_snapshot
```
Comment (ciejj) [inline app/models/order.rb]: I would consider it naming simply `snapshot` then you won't have calls like `order.order_snapshot` - simply `order.snapshot`

**Review by ciejj — APPROVED** [review]
Looks good - left some suggestions for minor updates

**Review by oskarlakner — APPROVED** [review]

File: app/models/employee.rb (hunk 1 above)
Comment (konrad04-1989) [reply app/models/employee.rb]: After the change I get the error `undefined method price`. After writing this method rubocop throws me the error and the proposal `delegate` and we are back to the same place again :)

File: app/models/order.rb (hunk 2 above)
Comment (konrad04-1989) [reply app/models/order.rb]: Revised. Thanks for the advice!

**Pull request merged**